iceoryx_hoofs/source/cxx/helplets.cpp

```diff
@@ -21,27 +21,33 @@ namespace iox
 {
 namespace cxx
 {
+// AXIVION DISABLE STYLE AutosarC++19_03-A5.2.4 : allow reinterpret_cast for the whole file
 void* alignedAlloc(const uint64_t alignment, const uint64_t size) noexcept
 {
     // -1 == since the max alignment addition is alignment - 1 otherwise the
     // memory is already aligned and we have to do nothing
     // low-level memory management
     // NOLINTNEXTLINE(cppcoreguidelines-owning-memory, cppcoreguidelines-no-malloc, hicpp-use-auto)
-    uint64_t memory = reinterpret_cast<uint64_t>(malloc(size + alignment + sizeof(void*) - 1));
+    // AXIVION Next Line AutosarC++19_03-M5.2.9 : Pointer conversion is safe here.
+    // AXIVION Next Line AutosarC++19_03-A18.5.1 : Dynamic memory management function is required.
+    const uint64_t memory = reinterpret_cast<uint64_t>(malloc((size + alignment + sizeof(void*)) - 1));
     if (memory == 0)
     {
         return nullptr;
     }
-    uint64_t alignedMemory = align(memory + sizeof(void*), alignment);
-    assert(alignedMemory >= memory + 1);
+    
+    const uint64_t alignedMemory = align(memory + sizeof(void*), alignment);
```

Comment:
This is a good point. Here I would suggest to implement `cxx::align` with `uint64_t` and perform overflow checks in there which should be easy since `uint64_t` overflow is at least defined behavior.

One philosophical thought of mine was to implement our own integer classes where one can define how overflow and underflow is handled, either by saturation, wrap around (UINT_MAX + 1 = 0) or fatal failure and use them everywhere in our code base.
The user code wouldn't have to change for this since we could make those classes convertible from and into their underlying integer type like we do with the relocatable pointer.